src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,383 @@
+
+    def __init__(
+        self,
+        cluster_name,
+        region,
+        boto3_config,
+        queue_config,
+        compute_resource_config,
+        all_or_nothing,
+    ):
+        super().__init__(
+            cluster_name,
+            region,
+            boto3_config,
+            queue_config,
+            compute_resource_config,
+            all_or_nothing,
+        )
+
+    def _evaluate_launch_params(self, count, launch_overrides):
+        """Evaluate parameters to be passed to run_instances call."""
+        launch_params = {
+            # Set MinCount to "count" to make the run_instances call fail if entire count cannot be satisfied
+            "MinCount": 1 if not self._all_or_nothing else count,
+            "MaxCount": count,
+            # LaunchTemplate is different for every compute resources in every queue
+            "LaunchTemplate": {
+                "LaunchTemplateName": f"{self._cluster_name}-{self._queue}-{self._compute_resource}",
+                "Version": "$Latest",
+            },
+        }
+
+        launch_params.update(launch_overrides)
+        if launch_overrides:
+            logger.info("Found RunInstances parameters override. Launching instances with: %s", launch_params)
+        return launch_params
+
+    def _launch_instances(self, launch_params):
+        """Launch a batch of ec2 instances."""
+        try:
+            return run_instances(self._region, self._boto3_config, launch_params)
+        except ClientError as e:
+            logger.error("Failed RunInstances request: %s", e.response.get("ResponseMetadata").get("RequestId"))
+            raise e
+
+
+class Ec2CreateFleetManager(FleetManager):
+    """Manager to create EC2 instances fleet using create_fleet API."""
+
+    def __init__(
+        self,
+        cluster_name,
+        region,
+        boto3_config,
+        queue_config,
+        compute_resource_config,
+        all_or_nothing,
+    ):
+        super().__init__(
+            cluster_name,
+            region,
+            boto3_config,
+            queue_config,
+            compute_resource_config,
+            all_or_nothing,
+        )
+
+    def _evaluate_launch_params(self, count, launch_overrides):
+        """Evaluate parameters to be passed to create_fleet call."""
+        overrides = []
+        for instance_type in self._compute_resource_config.get("InstanceTypeList", []):
+            overrides.append({"InstanceType": instance_type["InstanceType"]})
+
+        launch_params = {
+            "LaunchTemplateConfigs": [
+                {
+                    "LaunchTemplateSpecification": {
+                        # LaunchTemplate is different for every compute resources in every queue
+                        "LaunchTemplateName": f"{self._cluster_name}-{self._queue}-{self._compute_resource}",
+                        "Version": "$Latest",
+                    },
+                    "Overrides": overrides,
```

Comment:
missing the `LaunchTemplateConfigs/Overrides/MaxPrice` for the spot case. Correspond to the ParallelCluster configuration option [SpotPrice](https://docs.aws.amazon.com/parallelcluster/latest/ug/Scheduling-v3.html#yaml-Scheduling-SlurmQueues-ComputeResources-SpotPrice)

Reply:
Added. 

Manually tested config file with `SpotPrice: null` and with a value of 10. In the log I see `'Overrides': [{'MaxPrice': '10', 'InstanceType': 't2.medium'}]}],` if value is set, `'Overrides': [{'InstanceType': 't2.medium'}]}],` otherwise